File: 02-programacion-basica/09-colecciones/scala/practicas/PracticaInventarioColecciones.scala
@main def practicaInventarioColecciones(): Unit =
  var stock = Map("Teclado" -> 5, "Mouse" -> 8, "Monitor" -> 3)
  // updated crea un nuevo Map con el valor asociado a la clave reemplazado.
  stock = stock.updated("Mouse", stock("Mouse") - 2)
  stock.foreach { case (producto, cantidad) => println(s"$producto -> $cantidad unidades") }

File: 02-programacion-basica/09-colecciones/java/practicas/PracticaInventarioColecciones.java
import java.util.LinkedHashMap;
import java.util.Map;

public class PracticaInventarioColecciones {
    public static void main(String[] args) {
        // LinkedHashMap conserva el orden de inserción de las claves.
        Map<String, Integer> stock = new LinkedHashMap<>();
        stock.put("Teclado", 5);
        stock.put("Mouse", 8);
        stock.put("Monitor", 3);

        stock.replace("Mouse", stock.get("Mouse") - 2);

        for (var producto : stock.entrySet()) {
            System.out.printf("%s -> %d unidades%n", producto.getKey(), producto.getValue());
        }
    }
}
